File: verecorder/src/main/java/com/vesdk/verecorder/record/preview/model/Resolution.java
package com.vesdk.verecorder.record.preview.model;

/**
 * time : 2020/11/18
 *
 * description :
 */
public class Resolution {
    public String name;
    public int resolution;
    public float ratio;
    public int width;
    public int height;

    public Resolution(String name, int resolution) {
        this.name = name;
        this.resolution = resolution;
    }

    public Resolution(String name, int resolution, float ratio, int width, int height) {
        this.name = name;
        this.resolution = resolution;
        this.ratio = ratio;
        this.width = width;
        this.height = height;
    }

    public Resolution(String name, int resolution, float ratio) {
        this.name = name;
        this.resolution = resolution;
        this.ratio = ratio;
        this.width = resolution/ 16 * 16;
        this.height = ((int) (resolution * ratio)/ 16 * 16);
    }

    @Override
    public String toString() {
        return "name:"+name
                +" resolution:"+resolution
                +" ratio:"+ratio
                +" width:"+width
                +" height:"+height;
    }
}

File: editor-core/src/main/java/com/ss/ugc/android/editor/core/api/video/Resolution.kt
package com.ss.ugc.android.editor.core.api.video

/**
 * time : 2020/12/8
 *
 * description :
 *
 */
class Resolution(var resolution: Int, var ratio: Float) {

    var name: String? = ""
    var width: Int
    var height: Int

    init {
        this.width = resolution / 16 * 16
        this.height = (resolution / ratio).toInt() / 16 * 16
    }

    override fun toString(): String {
        return ("name:" + name
                + " resolution:" + resolution
                + " ratio:" + ratio
                + " width:" + width
                + " height:" + height)
    }
}